xonsh/tools.py

```diff
@@ -589,6 +600,38 @@ def is_bool(x):
     """Tests if something is a boolean."""
     return isinstance(x, bool)
 
+def is_tb_opt(x):
+    """
+    Tests if something is a valid XONSH_SHOW_TRACEBACK option, which can
+    be either a bool or a string containing a file path.
+    """
+    return isinstance(x, bool) or isinstance(x, str)
+
+def to_tb_tuple(x):
+    """
+    Converts a XONSH_SHOW_TRACEBACK option to a tuple containing:
+    -- a boolean indicating if traceback should be logged
+    -- a string option indicating the file it should be logged out to,
+       or None if no such option was provided.
+    """
+    if isinstance(x, bool):
+        return (x, None)
+    elif isinstance(x, str):
+        return (True, x.strip())
+    else:
+        raise ValueError('$XONSH_SHOW_TRACEBACK must either be a boolean '
+                         'or a string containing a file path.')
+
+def tb_tuple_to_str(x):
+    """Handles throwback opt detyping to a string."""
+    if isinstance(x, bool):
+        return str(x)
+    elif isinstance(x, str):
+        return '\'' + str(x) + '\''
```

Comment:
This should probably just return `repr(x)` instead of wrapping with quotes manually.